### PR TITLE
Upgrade Akeneo to V6

### PR DESCRIPTION
**Pull request by danielbeardsley**
Lots of iteration... even an embarrassing WIP commit.

Much of the config modifications come from running the upgrade script and committing the result.

Connect ifixit/ifixit#45026

### CODE REVIEW

**Comment by danielbeardsley** [comment]
QA :+1: This seems to operate cleanly and the boot process results in a working machine.

File: docker/Dockerfile [hunk 1]
```diff
@@ -2,11 +2,11 @@ FROM debian:buster-slim as base
...
 ENV PHP_CONF_DATE_TIMEZONE=UTC \
     PHP_CONF_MAX_EXECUTION_TIME=60 \
-    PHP_CONF_MEMORY_LIMIT=512M \
+    PHP_CONF_MEMORY_LIMIT=1G \
     PHP_CONF_OPCACHE_VALIDATE_TIMESTAMP=0 \
     PHP_CONF_MAX_INPUT_VARS=1000 \
-    PHP_CONF_UPLOAD_LIMIT=40M \
-    PHP_CONF_MAX_POST_SIZE=40M
+    PHP_CONF_UPLOAD_LIMIT=100M \
+    PHP_CONF_MAX_POST_SIZE=200M
```
Comment (masonmcelvain) [inline docker/Dockerfile:9]: We doubled the post max size here without a comment. Not sure that it matters but thought I'd point it out

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,8 +1,57 @@
...
+COPY ./build/akeneo.ini /etc/php/8.0/fpm/conf.d/99-akeneo.ini
+
+COPY --from=composer:2 /usr/bin/composer /usr/local/bin/composer
 RUN chmod +x /usr/local/bin/composer
-RUN mkdir -p /etc/ssl/certs/
-RUN apt-get update && apt-get install -y ca-certificates php-ds
 
-# USER 1000
+## There once was a bunch of extra stuff here to support the akeneo development
+## environment, but we don't need it cause we mostly *run* akeneo.
```
Comment (masonmcelvain) [inline docker/Dockerfile:55]: Are the `.env.dev` and dev docker-compose files unused then too?

File: Makefile [hunk 3]
```diff
@@ -131,9 +131,32 @@ up:
...
 
-.PHONY: upgrade
-upgrade: node_modules cache assets front-packages javascript-prod css javascript-extensions
+.PHONY: ifixit-upgrade
+ifixit-upgrade: dependencies cache assets front-packages javascript-prod css javascript-extensions
 	bash vendor/akeneo/pim-community-dev/std-build/install-required-files.sh
 	patch -p0 < patches/migrations.patch
 	cp .env .env.upgrade
 	cp .env.local .env
+	# Some migrations need elasticsearch to be running
```
Comment (masonmcelvain) [inline Makefile]: I'm not sure which boot script you're referring to in this commit 2fe044f4c4e0dac8f987bef3269d48982d0f3179

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 Everything seemed reasonable as far as I could understand. As long as it works 😅

File: docker/Dockerfile (hunk 1 above)
Comment (danielbeardsley) [reply docker/Dockerfile:9]: The docker file being updated was part of the upgrade script (it just copies it from pim-community-dev).

File: docker/Dockerfile (hunk 2 above)
Comment (danielbeardsley) [reply docker/Dockerfile:55]: No. little bit a misnomer here. We have a docker-compose-dev.yml and it's used to run akeneo (and mysql) in docker containers (so we can iterate in an isolated env). Akeneo "development" in this comment refers to actually editing and hacking on the pim-communinty-dev repo which we don't do (maintainers do).

File: Makefile (hunk 3 above)
Comment (danielbeardsley) [reply Makefile]: I apparently only linked it in one direction!

https://github.com/iFixit/server-templates/pull/4005/commits/8e7028124585906cfa7343331cb514e657826d60

**Comment by danielbeardsley** [comment]
QA :+1: 

This produces a working v6 Akeneo install and our integration still operates.

File: docker/Dockerfile (hunk 1 above)
Comment (masonmcelvain) [reply docker/Dockerfile:9]: I understand that most of this file was copied, but I think you may have hand edited line 9? I might be misunderstanding, but I thought the `akeneo-custom-php.ini` file was something custom we added. When we removed it in 992c960ec83e5d2fd4d3b5d9e41ceff0b1cc28a1, we replaced it with env vars, and we doubled the `post_max_size`. I wasn't sure if that was intentional, though it probably doesn't matter very much.

Comment (danielbeardsley) [reply docker/Dockerfile:9]: Hmm.. good point, I have a vague memory of changing it to 200M cause I thought I remembered it needing to be larger than UPLOAD_LIMIT. But you're right, it probably doesn't matter. I'm fine with leaving it at 200M

**Comment by danielbeardsley** [comment]
This is done! Akeneo v6 is live.

Further changes will merge into `ifixit-v6` branch.

**Pull request merged**